EIPS/eip-7743.md

```diff
@@ -0,0 +1,120 @@
+3. **Affordable**. Daily operations could be signed and executed with the local smart account operator. The ZK verification will only be used to reset the account if the operator key is lost or stolen to ensure the whole setup is affordable. ZK proofs aggregation has also proven to be viable and paves the way for cheaper verification cost at scale.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Definitions
+
+**Account**: The user's smart account contract
+
+**Operator**: The signer to sign and exectue transactions. This can be reset by the verifier if lost or stolen.
+
+**Identity Provider(IDP)**: The service to authenticate users and provide signed ID token
+
+**Client**: The application to authenticate users and generate the ZK proof
+
+**Aggregrator**: The offchain service to aggregate ZK proof from multiple users
+
+**Verifier**: The on-chain contract to verify the ZK proof and recover account by resetting the operator
+
+### Example Flow
+
+![The example workflow](../assets/eip-7743/workflow.png)
+
+### Interface
+
+```
+interface IOpenIdVerifier {
+    struct Action {
+        address account;
+        bytes data;
+    }
+ 
+    // @notice get verification key of the open id authenticator
+    function getVerificationKey() external view returns(bytes memory);
+ 
+    // @notice get id hash of account
+    function getIdHash(address account) external view returns(bytes32);
+
+    // @notice the function verifies the proof and executes the action if it's valid
+    // @params action: the action to execute
+    //         proof: data to prove the account ownership
+    function verifyAndCall(
+        Action memory action,
+        bytes memory proof
+    ) external;
+
+    // @notice the function verifies the aggregated proof and
+    //         execute actions in batch if it's valid
+    // @params actions: a list of actions to execute
+    //         aggregatedProof: the data to prove the ownership of accounts in actions
+    function verifyAndCallAggregated(
+        Action[] memory actions,
+        bytes memory aggregatedProof
+    ) external;
+}
+
+interface IOpenIdAccount {
+    /// @notice get open id verifier linked to the account
+    /// @return the open id verifier
+    function getOpenIdVerifier() external view returns(IOpenIdVerifier);
+}
+
+interface IRecoverableAccount is IOpenIdAccount {
+    /// @notice get the operator of the account
+    /// @return the address of operator
+    function getOperator() external view returns(address);
+ 
+    /// @notice set operator of the account
+    /// @params the new operator to set
+    function setOperator(address operator) external;
```

Comment:
[Security] does it imply giving whoever could `setOperator` a very high privilegde? Should the setter get compromise, regardless of it being `IOpenIdVerifier` or other kind of contract accounts, is it additional risk?

[Technical Feedback] It seems the setOperator could be achieved by https://eips.ethereum.org/EIPS/eip-173 Ownable's `transferOwnership` or https://eips.ethereum.org/EIPS/eip-5982 for grantRole if needs more flexible ACL

Reply:
1. It really depends on how the account contract use it. This EIP is more about the IOpenIdVerifier not the account implementation. One may have very complicated account implementation with better permission control over setOperator function.
2. yeah I think ERC173 is sufficient as the example, will update the interface to inherit from ERC173 directly.

Reply:
For the first question, remember that IOpenIdVerifier is permissionless so the only way to compromise it is to hack your OpenId identity. One common enhancement is to add additional verifiers so the account can only be hacked if all linked open ids are hacked, or to add extra locking period to ensure account ownership cannot be transferred for a period of time.